lib/task_listener.js

```diff
@@ -77,6 +77,11 @@ export default class TaskListener extends EventEmitter {
   }
 
   async listenForCancelEvents() {
+    // Do not listen for events if pulse configuration is not supplied
```

Comment:
This is mainly for when the worker for some reason restarts and can't retrieve the creds from secrets because they were removed.  Capacity is set to 0 and we expect the worker to die off, so it should not need to listen for cancel events anyways.